ooni/nettests/blocking/web_connectivity.py

```diff
@@ -176,9 +172,13 @@ def setUp(self):
         self.report['tcp_connect'] = []
         self.report['control'] = {}
 
-        self.hostname = urlparse(self.input).netloc
-        if not self.hostname:
-            raise AbsentHostname('No hostname', self.input)
+        url = prepend_scheme_if_missing(self.input)
```

Comment:
Since we are already doing this in the input_processor, to avoid doing this twice, this should instead be run only when we pass the `url` option (i.e. put this function call at line 147 `self.input = prepend_scheme_if_missing (self.localOptions['url'])`)